spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-yarn/src/main/java/org/springframework/cloud/data/module/deployer/yarn/YarnModuleDeployer.java

```diff
@@ -0,0 +1,213 @@
+
+		// Using same app instance yarn boot cli is using to
+		// communicate with an app running on yarn via its boot actuator
+		YarnContainerClusterApplication app = new YarnContainerClusterApplication();
+		Properties appProperties = new Properties();
+		appProperties.setProperty(PREFIX + "operation", "CLUSTERCREATE");
+		appProperties.setProperty(PREFIX + "applicationId", yarnApplicationId);
+		appProperties.setProperty(PREFIX + "clusterId", clusterId);
+		appProperties.setProperty(PREFIX + "clusterDef", "module-template");
+		appProperties.setProperty(PREFIX + "projectionType", "default");
+		appProperties.setProperty(PREFIX + "projectionData.any", Integer.toString(count));
+		appProperties.setProperty(PREFIX + "extraProperties.containerModules", module);
+		app.appProperties(appProperties);
+		String output = app.run();
+		logger.info("Output from YarnContainerClusterApplication run for CLUSTERCREATE: " + output);
+
+		app = new YarnContainerClusterApplication();
+		appProperties = new Properties();
+		appProperties.setProperty(PREFIX + "operation", "CLUSTERSTART");
+		appProperties.setProperty(PREFIX + "applicationId", yarnApplicationId);
+		appProperties.setProperty(PREFIX + "clusterId", clusterId);
+		app.appProperties(appProperties);
+		output = app.run();
+		logger.info("Output from YarnContainerClusterApplication run for CLUSTERSTART: " + output);
+
+		return moduleDeploymentId;
+	}
+
+	@Override
+	public void undeploy(ModuleDeploymentId id) {
+		String clusterId = convertFromModuleDeploymentId(id);
+		String yarnApplicationId = findRunningXdYarnApp();
+		YarnContainerClusterApplication app = new YarnContainerClusterApplication();
+		Properties appProperties = new Properties();
+		appProperties.setProperty(PREFIX + "operation", "CLUSTERSTOP");
+		appProperties.setProperty(PREFIX + "applicationId", yarnApplicationId);
+		appProperties.setProperty(PREFIX + "clusterId", clusterId);
+		app.appProperties(appProperties);
+		String output = app.run();
+		logger.info("Output from YarnContainerClusterApplication run for CLUSTERSTOP: " + output);
+	}
+
+	@Override
+	public ModuleStatus status(ModuleDeploymentId id) {
+		String yarnApplicationId = findRunningXdYarnApp();
+		String clusterId = convertFromModuleDeploymentId(id);
+
+		YarnContainerClusterApplication app = new YarnContainerClusterApplication();
+		Properties appProperties = new Properties();
+		appProperties.setProperty(PREFIX + "operation", "CLUSTERINFO");
+		appProperties.setProperty(PREFIX + "applicationId", yarnApplicationId);
+		appProperties.setProperty(PREFIX + "verbose", "false");
+		appProperties.setProperty(PREFIX + "clusterId", clusterId);
+		app.appProperties(appProperties);
+		String info = app.run();
+		logger.info("Output from YarnContainerClusterApplication run for CLUSTERINFO: " + info);
+
+		boolean deployed = false;
+		String[] lines = info.trim().split("\\r?\\n");
+		if (lines.length == 3 && lines[2].contains("RUNNING")) {
```

Comment:
Are there any other possible statuses? For instance, if an exception is thrown will the output say ERROR or something like that? For now it looks like the only possible statuses are DEPLOYED and UNKNOWN. If Yarn blocks on every invocation until the application is running, it appears that a single module will never be in a DEPLOYING state.